src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs

```diff
@@ -65,26 +65,27 @@ public async Task AnalyzeAsync(IProject project, IDependencyAnalysisState state,
                     }
                     else
                     {
-                        _logger.LogInformation("Marking package {NuGetPackage} for removal because it doesn't support the target framework but a newer version ({Version}) does", packageReference, updatedReference.Version);
+                        var logMessage = SR.Format("Package {0} does not support the target(s) {1} but a newer version ({2}) does.", packageReference, string.Join(", ", state.TargetFrameworks), updatedReference.Version);
+                        _logger.LogInformation(logMessage);
                         var isMajorChange = _comparer.IsMajorChange(updatedReference.Version, packageReference.Version);
 
                         if (isMajorChange)
                         {
-                            var logString = SR.Format("Package {0} needs to be upgraded across major versions ({1} -> {2}) which may introduce breaking changes", packageReference.Name, packageReference.Version, updatedReference.Version);
-                            details.Add(logString);
+                            var logString = SR.Format("Package {0} needs to be upgraded across major versions ({1} -> {2}) which may introduce breaking changes.", packageReference.Name, packageReference.Version, updatedReference.Version);
+                            details.Add(string.Concat(logMessage, logString));
```

Comment:
Do you want to put a <space> between the two? Maybe interpolation would work?

Reply:
Good point, changed to use interpolation.